### PR TITLE
changed secret lcd and added more control flow and error handling

### PR DESCRIPTION
**Pull request by Ionut33**
1. Changed LCD url
2. Added error handling
3. Control flow (rate limit)

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/sienna exports TVL: 
        
 
 
    secret-staking            4.04 M
    secret                    12.80 M
    staking                   4.04 M
    
    total                    12.80 M

**Comment by llamatester** [comment]
The adapter at projects/sienna exports TVL: 
        
 
 
    secret-staking            4.41 M
    secret                    13.36 M
    staking                   4.41 M
    
    total                    13.36 M

File: projects/sienna/index.js [hunk 1]
```diff
@@ -16,23 +20,25 @@ const SIENNA_SINGLE_SIDED_POOLS = [
...
-async function PairsV2() {
-    const pairs = await queryClient.getContracts(pairCodeIdV2);
-    return pairs.filter((p) => p.label.endsWith(`${factoryContractV2}-${pairCodeIdV2}`));
+    return new Promise((resolve) => {
+        mapLimit(factories, 1, async (factory) => {
+            try {
+                const result = await queryClient.getContracts(factory.code);
+                const pairs = result.filter((p) => p.label.endsWith(`${factory.address}-${factory.code}`));
+                return Promise.resolve(pairs);
+            } catch (e) {
```
Comment (g1nt0ki) [inline projects/sienna/index.js:35]: Hi, can you revert the error handling, we prefer adapter failing over returning wrong tvl

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
Fail the adapter when there is an error

**Comment by llamatester** [comment]
The adapter at projects/sienna exports TVL: 
        
 
 
    secret-staking            4.43 M
    secret                    13.36 M
    staking                   4.43 M
    
    total                    13.36 M

File: projects/sienna/index.js (hunk 1 above)
Comment (Ionut33) [reply projects/sienna/index.js]: Understood, reverted the error handling

File: projects/sienna/index.js [hunk 2]
```diff
@@ -137,7 +137,7 @@ async function StakedTokens() {
             }
             const tokens = new BigNumber(total_locked).div(new BigNumber(10).pow(siennaToken.decimals)).toNumber();
             return Promise.resolve(tokens);
-        }, (err, data) => {
+        }, (_err, data) => {
```
Comment (g1nt0ki) [inline projects/sienna/index.js:140]: can you please add a reject for _err here

Comment (Ionut33) [reply projects/sienna/index.js]: Yes, now all errors are thrown out of the async loops

File: projects/sienna/index.js (hunk 1 above)
Comment (g1nt0ki) [reply projects/sienna/index.js]: thanks for fixing it quickly

**Pull request merged**